zaza/openstack/charm_tests/ceph/dashboard/tests.py

```diff
@@ -309,3 +311,115 @@ def test_saml(self):
+        server_cert = local_ca.issue_cert(
+            "ceph-dashboard", key_type=trustme.KeyType.RSA)
+
+        ssl_cert = b64encode(server_cert.cert_chain_pems[0].bytes()).decode()
+        ssl_key = b64encode(server_cert.private_key_pem.bytes()).decode()
+        ssl_ca = b64encode(local_ca.cert_pem.bytes()).decode()
+
+        # Configure local certs in charm config
+        zaza.model.set_application_config(
+            'ceph-dashboard',
+            {
+                'ssl_cert': ssl_cert, 'ssl_key': ssl_key,
+                'ssl_ca': ssl_ca
+            }
+        )
+
+        # Check application status message.
+        assert_state = self._get_wait_for_dashboard_assert_state(
+            "blocked", "SSL configured from 'certificates' relation."
+        )
+        zaza.model.wait_for_application_states(
+            states=assert_state, timeout=500
+        )
+
+        # Remove certificates relation to trigger configured certs.
+        zaza.model.remove_relation(
+            'ceph-dashboard', 'ceph-dashboard:certificates',
+            'vault:certificates'
+        )
+
+        # Wait for status to clear
+        assert_state = self._get_wait_for_dashboard_assert_state(
+            "active", "Unit is ready"
+        )
+        zaza.model.wait_for_application_states(
+            states=assert_state, timeout=500
+        )
+
+        # Verify Certificates.
+        rcs = collections.defaultdict(list)
+        units = zaza.model.get_units('ceph-mon')
+        with local_ca.cert_pem.tempfile() as ca_temp_file:
+            for unit in units:
+                r = self._run_request_get(
+                    'https://{}:8443'.format(
+                        zaza.model.get_unit_public_address(unit)),
+                    verify=ca_temp_file,
+                    allow_redirects=False)
+                rcs[r.status_code].append(
+                    zaza.model.get_unit_public_address(unit)
+                )
+        self.assertEqual(len(rcs[requests.codes.ok]), 1)
+        self.assertEqual(len(rcs[requests.codes.see_other]), len(units) - 1)
+
+        # Re-add certificates relation
+        zaza.model.add_relation(
+            'ceph-dashboard', 'ceph-dashboard:certificates',
+            'vault:certificates'
+        )
+
+        # Check blocked status message
+        assert_state = self._get_wait_for_dashboard_assert_state(
+            "blocked", "SSL configured from Charm config."
+        )
+        zaza.model.wait_for_application_states(
+            states=assert_state, timeout=500
+        )
+
+        # Remove SSL config
+        zaza.model.set_application_config(
+            'ceph-dashboard',
+            {'ssl_cert': "", 'ssl_key': "", 'ssl_ca': ""}
+        )
+
+        # Wait for status to clear
+        assert_state = self._get_wait_for_dashboard_assert_state(
+            "active", "Unit is ready"
+        )
+        zaza.model.wait_for_application_states(
+            states=assert_state, timeout=500
+        )
```

Comment:
I'm wondering if it would make sense to run another round of self._run_request_get() here in order to verify SSL works correctly again? (e.g. put the certificate verification section above from L383-L397 into a helper method and call it two times)

Reply:
ack